fix: Fixes Update SDK GitHub action

lantoli opened the pull request:
## Description

Jira ticket: [INTMDB-1252](https://jira.mongodb.org/browse/INTMDB-1252)

Fixes Update SDK GitHub action. 

- Uses a token to create the PR so CI checks are run
- Removes file filters to include all changed files.

Example of PR created: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1599



## Type of change:

- [X] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [X] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [X] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

## Further comments

lantoli committed:
include all changed files

lantoli committed:
install gomajor from make tools

lantoli committed:
allow to use SDK Preview

lantoli committed:
use Token so CI checks are run

lantoli committed:
fix sdk preview script

wtrocki commented on .github/workflows/update-sdk.yml:
```diff
@@ -13,23 +13,18 @@ jobs:
-          go install github.com/icholy/gomajor@latest 
-          make update-atlas-sdk
+      - name: Update files
+        run:  make tools update-atlas-sdk
```
[Nit] Change adds to the build time. Ok to leave it.

lantoli replied:
i know, but just a few seconds as a tradeoff for consistency and remove duplication

wtrocki commented on .github/workflows/update-sdk.yml:
```diff
@@ -13,23 +13,18 @@ jobs:
+        run:  make tools update-atlas-sdk
       - name: Verify Changed files
-        uses: tj-actions/verify-changed-files@6d688963a73d28584e163b6f62cf927a282c4d11
+        uses: tj-actions/verify-changed-files@v16
```
We do pin actions to commit. More info:
https://medium.com/@wtr/best-patterns-building-github-actions-workflows-6b03655223ca

```suggestion
        uses: tj-actions/verify-changed-files@6d688963a73d28584e163b6f62cf927a282c4d11
```

lantoli replied:
I see

lantoli committed:
Update verify-changed-files to a commit version

Co-authored-by: Wojciech Trocki <w.trocki@mongodb.com>

wtrocki commented:
We need to verify if action changes work. Best would be to trigger it [from branch](https://medium.com/@wtr/best-patterns-building-github-actions-workflows-6b03655223ca)

lantoli commented:
> We need to verify if action changes work. Best would be to trigger it [from branch](https://medium.com/@wtr/best-patterns-building-github-actions-workflows-6b03655223ca)

@wtrocki I did a few tests from this branch

wtrocki commented on scripts/update-sdk-preview.sh:
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
```
```suggestion
set -euo pipefail

echo "Script will update to dev preview. Note: you need to have latest major version in order to be able to use it"
```

wtrocki commented:
Verified it myself. All good. Added extra note as script might not work for all the cases

lantoli committed:
Update text in  scripts/update-sdk-preview.sh

Co-authored-by: Wojciech Trocki <w.trocki@mongodb.com>

lantoli commented:
> Verified it myself. All good. Added extra note as script might not work for all the cases

@wtrocki added your note. that's also why I added this note: "If no files were changed you may need first to update to the latest SDK using the 'Update SDK' GitHub Action"

lantoli committed:
change doc

andreaangiolillo reviewed (changes requested):
Why do we want to use SDK preview? Doesn't SDK preview use cloud-dev?

marcosuma commented on GNUmakefile:
```diff
@@ -138,3 +139,7 @@ link-git-hooks: ## Install git hooks
 	./scripts/update-sdk.sh
+
+.PHONY: try-atlas-sdk-preview
+try-atlas-sdk-preview: ## Use SDK Preview, DONT MERGE
```
If we really have to push this, this needs some sort of automated safety mechanism that would impede a push in master where we're using the sdk preview version. I can see how with a distraction we can incur in unexpected merges.

marcosuma replied:
in the meantime if your purpose is to test search index you can unblock yourself by testing this locally and syncing with the owning team on when is the plan of releasing these APIs in prod. I don't think we should tweak things and build workarounds to project timeline obstacles.

lantoli replied:
@andreaangiolillo @marcosuma as @wtrocki is saying, this is to make it easier for people what wants/need to use the SDK Preview.   
In this way people don't need to do it manually. what problems do you see keeping this make target in case somebody wants to use it?  (there can always be a distraction and merge sdk preview changes even if this make target doesn't exist)

andreaangiolillo replied:
> what problems do you see keeping this make target in case somebody wants to use it?

We are adding additional code to maintain for something that I think we should not use. If we need to use the SDK-preview, a better way which does not involve updating all the repo files is to use `replace =>` with the local SDK pointing to the SDK preview PR.

lantoli replied:
only go.mod and go.sum are modified when using the SDK Preview

andreaangiolillo replied:
Yes, it is true. One additional question: What would happen if the preview PR is not available on the SDK channel when running the script? Would the script fail, would the script run the github action on the atlas SDK?

I think you can move the preview logic to a new  PR so that we can keep discussing it there and unblock this PR

wtrocki commented:
> Why do we want to use SDK preview? Doesn't SDK preview use cloud-dev?

The idea is to do development of the feature using a preview before the change is available. 

EDIT: closed by mistake.

lantoli committed:
remove preview

lantoli committed:
more descriptive action name

andreaangiolillo reviewed (approved):
LGTM. Thanks for the discussions
